hoot-core/src/main/cpp/hoot/core/conflate/review/ReviewMarker.cpp

```diff
@@ -159,58 +159,40 @@ bool ReviewMarker::isReviewUid(const ConstOsmMapPtr &map, ReviewUid uid)
   return isReview(map->getElement(uid));
 }
 
-// TODO: consolidate the duplicate code in these mark methods, if possible
-
 void ReviewMarker::mark(const OsmMapPtr &map, const ElementPtr& e1, const ElementPtr& e2,
   const QString& note, const QString& reviewType, double score, vector<QString> choices)
 {
-  LOG_TRACE("Marking review...");
-
-  LOG_VART(reviewType);
-  LOG_VART(note);
-  LOG_VART(score);
-
-  if (note.isEmpty())
-  {
-    LOG_VART(e1->toString());
-    LOG_VART(e2->toString());
-    throw IllegalArgumentException("You must specify a review note.");
-  }
-
-  RelationPtr r(
-    new Relation(
-      Status::Conflated, map->createNextRelationId(), 0, MetadataTags::RelationReview()));
-  r->getTags().set(MetadataTags::HootReviewNeeds(), true);
-  if (_addReviewTagsToFeatures)
-  {
-    e1->getTags().set(MetadataTags::HootReviewNeeds(), true);
-    e2->getTags().set(MetadataTags::HootReviewNeeds(), true);
-  }
-  r->getTags().appendValueIfUnique(MetadataTags::HootReviewType(), reviewType);
-  if (ConfigOptions().getWriterIncludeConflateReviewDetailTags())
-  {
-    r->getTags().appendValueIfUnique(MetadataTags::HootReviewNote(), note.simplified());
-    r->getTags().set(MetadataTags::HootReviewScore(), score);
-  }
-  r->addElement(MetadataTags::RoleReviewee(), e1->getElementId());
-  r->addElement(MetadataTags::RoleReviewee(), e2->getElementId());
-  r->getTags().set(MetadataTags::HootReviewMembers(), (int)r->getMembers().size());
-  r->setCircularError(ElementData::CIRCULAR_ERROR_EMPTY);
-
-  LOG_VART(r->getId());
-  LOG_VART(e1->getElementId());
-  LOG_VART(e2->getElementId());
+  if (!e1 || !e2)
+    return;
+  //  Insert the IDs into a vector and call mark
+  vector<ElementId> ids;
+  ids.push_back(e1->getElementId());
+  ids.push_back(e2->getElementId());
+  mark(map, ids, note, reviewType, score, choices);
+}
```

Comment:
Each of the overloads just puts the `ElementId` for the parameter(s) into a `std::vector` and passed off to the old `std::set` version of the function and the code is reused.